vochain/transaction.go

```diff
@@ -295,23 +298,21 @@ func AdminTxCheck(vtx *models.Tx, txBytes, signature []byte, state *State) error
 		if !process.EnvelopeType.EncryptedVotes && !process.EnvelopeType.Anonymous {
 			return fmt.Errorf("process does not require keys")
 		}
-		// get the current blockchain header
-		header := state.Header(false)
-		if header == nil {
-			return fmt.Errorf("cannot get blockchain header")
-		}
+
+		height := state.Height()
 		// Specific checks
 		switch tx.Txtype {
 		case models.TxType_ADD_PROCESS_KEYS:
 			if tx.KeyIndex == nil {
 				return fmt.Errorf("missing keyIndex on AdminTxCheck")
 			}
 			// endblock is always greater than start block so that case is also included here
-			if header.Height > int64(process.StartBlock) {
+			if height > process.StartBlock {
 				return fmt.Errorf("cannot add process keys in a started or finished process")
 			}
 			// process is not canceled
-			if process.Status == models.ProcessStatus_CANCELED || process.Status == models.ProcessStatus_ENDED || process.Status == models.ProcessStatus_RESULTS {
+			if process.Status == models.ProcessStatus_CANCELED || process.Status == models.ProcessStatus_ENDED ||
+				process.Status == models.ProcessStatus_RESULTS {
 				return fmt.Errorf("cannot add process keys in a canceled process")
```

Comment:
this error message can be misleading if the process status is ENDED or RESULTS, maybe we can return the process status formatted on the error

Reply:
Well that's not from this commit but I'll fix it. Thanks.